crates/sui-indexer/src/handlers/objects_snapshot_processor.rs

```diff
@@ -113,23 +118,27 @@ where
         // While the below is true, we are in backfill mode, and so `ObjectsSnapshotProcessor` will
         // no-op. Once we exit the loop, this task will then be responsible for updating the
         // `objects_snapshot` table.
-        while latest_fn_cp > start_cp + self.config.snapshot_max_lag as u64 {
-            tokio::select! {
-                _ = self.cancel.cancelled() => {
-                    info!("Shutdown signal received, terminating object snapshot processor");
-                    return Ok(());
-                }
-                _ = tokio::time::sleep(std::time::Duration::from_secs(self.config.sleep_duration)) => {
-                    latest_fn_cp = self.client.get_latest_checkpoint().await?.sequence_number;
-                    start_cp = self
-                        .store
-                        .get_latest_object_snapshot_checkpoint_sequence_number()
-                        .await?
-                        .unwrap_or_default();
+        if latest_snapshot_cp == latest_indexer_cp {
```

Comment:
Ok I think I get it now, this condition is basically while we are in backfill mode

Reply:
In other words
While snapshot cp = max indexer cp

Stay in here until we are within checkpoint lag

Reply:
When we break out we are now using objects snapshot processor